client/src/game/store.ts

```diff
@@ -244,6 +246,32 @@ class GameStore extends VuexModule implements GameState {
         if (data.sync) socket.emit("Note.New", data.note);
     }
 
+    @Mutation
+    newMarker(data: { marker: string; sync: boolean }): void {
+        const exists = this.markers.some(m => m === data.marker);
+        if (!exists) {
+            this.markers.push(data.marker);
+            if (data.sync) socket.emit("Marker.New", data.marker);
+        }
+    }
+
+    @Mutation
+    removeMarker(data: { marker: string; sync: boolean }): void {
+        this.markers = this.markers.filter(m => m !== data.marker);
+        if (data.sync) socket.emit("Marker.Remove", data.marker);
+    }
+
+    @Mutation
+    jumpToMarker(marker: string): void {
+        const shape = layerManager.UUIDMap.get(marker);
```

Comment:
You should handle the case where the shape does not exist.

Reply:
Not sure what you mean here.... in which function?

Reply:
Got it. Updated now.